File: trunk/src/dotnet/yaxgl/YaxglLibrary/Controls/EditBox.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace de.yaxgl
{
    public class EditBox :Control
    {
        public EditBox(Container owner, string ID)
        {
            this.owner = owner;
            this.ID = ID;

            this.control = new System.Windows.Forms.TextBox();
            this.control.Click += new System.EventHandler(clickEvent);
            this.control.GotFocus += new System.EventHandler(gotFocusEvent);
            this.control.LostFocus += new System.EventHandler(lostFocusEvent);
        }

        public override void  initializeNativeControl(System.Xml.XmlElement xmlElement)
        {
            setBounds(Convert.ToInt32(xmlElement.Attributes["xpos"].InnerText),
                      Convert.ToInt32(xmlElement.Attributes["ypos"].InnerText),
                      Convert.ToInt32(xmlElement.Attributes["width"].InnerText),
                      Convert.ToInt32(xmlElement.Attributes["height"].InnerText));
            setText(xmlElement.Attributes["text"].InnerText);
            setMaxLength(Convert.ToInt32(xmlElement.Attributes["maxlength"].InnerText));
        }
    
        
        
        public void setText(string text)
        {
            this.control.Text = text;
        }

        public string getText()
        {
            return this.control.Text;
        }
        
        public void setMaxLength(int length) 
        {
            ((System.Windows.Forms.TextBox)this.control).MaxLength = length;
        
        }

        public int getMaxLength()
        {
            return  ((System.Windows.Forms.TextBox)this.control).MaxLength;
        }
    }
}

File: src/gwt/yaxgl-gwt-lib/src/de/yaxgl/client/Controls/EditBox.java
package de.yaxgl.client.Controls;

import com.google.gwt.xml.client.Element;

import de.yaxgl.client.Base.Control;
import de.yaxgl.client.Container.Container;
import de.yaxgl.client.EventListener.ClickListener;
import de.yaxgl.client.EventListener.FocusListener;

public class EditBox extends Control {

	
	public EditBox(Container owner,String ID)
	{
		this.owner=owner;
		this.ID=ID;
		this.control = new com.google.gwt.user.client.ui.TextBox();
		
		control.addStyleName("yaxgl-editbox");
		
		owner.add(this);
	}
	
	public void initializeNativeControl(Element xmlElement) {

		setBounds(Integer.valueOf(xmlElement.getAttribute("xpos")),
				Integer.valueOf(xmlElement.getAttribute("ypos")),
				Integer.valueOf(xmlElement.getAttribute("width")),
				Integer.valueOf(xmlElement.getAttribute("height")));
		
		 setText(xmlElement.getAttribute("text"));
         //setMaxLength(Integer.valueOf(xmlElement.getAttribute("maxlength")).intValue());
	}

    public void setText(String text)
    {
    	((com.google.gwt.user.client.ui.TextBox)this.control).setText(text);
    }

    public String getText()
    {
        return ((com.google.gwt.user.client.ui.TextBox)this.control).getText();
    }
    
    public void setMaxLength(int length) 
    {
    	((com.google.gwt.user.client.ui.TextBox)this.control).setMaxLength(length);
    }

    public int getMaxLength()
    {
    	return ((com.google.gwt.user.client.ui.TextBox)this.control).getMaxLength();
    }
    
    
    
    /* methods for dealing with the EventListener Interfaces */
    public void addClickListener(ClickListener clickListener) {
   	 ((com.google.gwt.user.client.ui.TextBox)control).addClickListener(clickListener);
    }
    public void removeClickListener(ClickListener clickListener) {
   	 ((com.google.gwt.user.client.ui.TextBox)control).removeClickListener(clickListener);
    }
    public void addFocusListener(FocusListener focusListener) {
   	 ((com.google.gwt.user.client.ui.TextBox)control).addFocusListener(focusListener);
    }
    public void removeFocusListener(FocusListener focusListener) {
   	 ((com.google.gwt.user.client.ui.TextBox)control).removeFocusListener(focusListener);
    }
    
}
